### PR TITLE
BidTree storage

### PR DESCRIPTION
**Pull request by vlopes11**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,3 +16,8 @@ thiserror = "1.0.20"
 rand_core = "0.5.1"
 lazy_static = "1.4.0"
 rand = "0.7"
+nstack = "0.5"
+
+[dependencies.poseidon252]
+git = "https://github.com/dusk-network/Poseidon252"
+branch = "vlopes11/poseidon-tree-annot-generic"
```
Comment (CPerezz) [inline Cargo.toml:23]: Bump poseidon first and use a tag.

File: Cargo.toml [hunk 2]
```diff
@@ -4,12 +4,10 @@ version = "0.2.1"
 authors = ["CPerezz <carlos@dusk.network>", "Victor Lopez <victor@dusk.network>"]
 edition = "2018"
 
-
 [dependencies]
 anyhow = "1.0.0"
 dusk-pki = {git = "https://github.com/dusk-network/dusk-pki", tag = "v0.1.1"}
```
Comment (CPerezz) [inline Cargo.toml:9]: use `v0.2.0`

**Review by CPerezz — CHANGES_REQUESTED** [review]
Also, bump the version in the PR and publish a release once it's merged please :)

Good job with the PR, just need to fix the `dusk-analyzer` lints!

File: Cargo.toml (hunk 1 above)
Comment (vlopes11) [reply Cargo.toml]: Done

File: Cargo.toml (hunk 2 above)
Comment (vlopes11) [reply Cargo.toml]: Done

**Review by CPerezz — APPROVED** [review]
LGTM

Really Good job!!

**Pull request merged**